File: vitrivr-engine-module-m3d/src/main/java/org/vitrivr/engine/model3d/data/render/lwjgl/render/RenderOptions.java

package org.vitrivr.engine.model3d.data.render.lwjgl.render;

import java.io.Serializable;
import java.util.function.Function;
import org.joml.Vector4f;

/**
 * RenderOptions
 * <ul>
 * <li>Used to switch on or off the texture rendering</li>
 * <li>Used to switch on or off the coloring rendering</li>
 * <li>Returns the color for the given value</li>
 * <li>Can be used to colorize the model custom</li>
 * </ul>
 */
public class RenderOptions implements Serializable {

  /**
   * Used to switch on or off the texture rendering
   */
  public boolean showTextures = true;

  /**
   * Used to switch on or off the coloring rendering For future face coloring
   */
  @SuppressWarnings("unused")
  public boolean showColor = false;

  /**
   * Returns the color for the given value Can be used to colorize the model custom
   *
   * @TODO: This cannot be serialized!
   */
  public transient Function<Float, Vector4f> colorfunction = (v) -> new Vector4f(v, v, v, 1f);
}

File: vitrivr-engine-module-m3d/src/main/kotlin/org/vitrivr/engine/model3d/lwjglrender/render/RenderOptions.kt
package org.vitrivr.engine.model3d.lwjglrender.render

import org.vitrivr.engine.core.model.mesh.texturemodel.util.types.Vec4f
import java.io.Serializable
import java.util.function.Function

/**
 * RenderOptions
 *
 *  * Used to switch on or off the texture rendering
 *  * Used to switch on or off the coloring rendering
 *  * Returns the color for the given value
 *  * Can be used to colorize the model custom
 */
data class RenderOptions(
    /**
     * Used to switch on or off the texture rendering
     */
    var showTextures: Boolean = true,

    /**
     * Used to switch on or off the coloring rendering For future face coloring
     */
    var showColor: Boolean = false,

    /**
     * Returns the color for the given value Can be used to colorize the model custom
     *
     * @TODO: This cannot be serialized!
     */
    @Transient
    var colorfunction: Function<Float, Vec4f> = Function { v: Float? -> Vec4f(v!!, v, v, 1f) }
) : Serializable
